homeassistant/components/tplink/entity.py

```diff
@@ -20,10 +21,14 @@
 def async_refresh_after(
     func: Callable[Concatenate[_T, _P], Awaitable[None]],
 ) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, None]]:
-    """Define a wrapper to refresh after."""
+    """Define a wrapper to catch exceptions and refresh after calls."""
 
     async def _async_wrap(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
-        await func(self, *args, **kwargs)
+        try:
+            await func(self, *args, **kwargs)
+        except SmartDeviceException as ex:
```

Comment:
I think we should catch `AuthenticationException` first and raise `ConfigEntryAuthFailed` exception.

Reply:
This is already done by the recent commit.